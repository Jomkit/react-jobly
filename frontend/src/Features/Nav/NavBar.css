nav {
    background-color: aliceblue;
    margin: auto;
}

nav a {
    text-decoration: none;
    color: white;
    margin-right: 10px;
}

nav a.active {
    color: rgb(105, 105, 236);
}