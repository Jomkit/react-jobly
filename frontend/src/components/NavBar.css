nav {
    background-color: darkgoldenrod;
    margin: auto;
}

nav a {
    text-decoration: none;
    color: white;
    margin-right: 10px;
}

nav a:active {
    color: grey;
}