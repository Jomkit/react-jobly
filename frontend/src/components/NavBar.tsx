import { NavLink } from 'react-router-dom'
import "./NavBar.css";
import { Nav, NavItem, Navbar } from 'reactstrap';
import { useContext } from 'react';
import { userContext } from './contexts/userContext';
import LogoutForm from './LogoutForm';

const NavBar = () => {
  const currUser = useContext(userContext);
  return (
    <Navbar color='info' dark>
          <Nav className='d-flex align-items-center'>
            <NavItem>
              <NavLink className="navbar-brand" to="/">React-Jobly</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/companies">Companies</NavLink>
            </NavItem>
            <NavItem>
                <NavLink to="/jobs">Jobs</NavLink>
            </NavItem>
          </Nav>
            <NavItem style={{ listStyle: "none" }}>
              { currUser ? 
                  <div className='d-flex align-items-center'>
                    <NavLink to="/profile">Profile</NavLink>
                    <LogoutForm />
                  </div>
                :
                  <>
                    <NavLink to="/login">Login</NavLink>
                    <NavLink to="/signup">Sign Up</NavLink>
                  </>
              }
            </NavItem>
      </Navbar>
  )
}

export default NavBar