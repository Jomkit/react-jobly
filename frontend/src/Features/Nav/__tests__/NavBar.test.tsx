import { render } from "@testing-library/react"
import NavBar from "../NavBar"
import { MemoryRouter } from "react-router-dom"

const TestNavBar = () => {
    return (
        <MemoryRouter>
            <NavBar />
        </MemoryRouter>
    )
}

it('should render without crashing', () => { 
    render(<TestNavBar />);
})

it('should match snapshot', () => { 
    const { asFragment } = render(<TestNavBar />);
    expect(asFragment()).toMatchSnapshot();  
})