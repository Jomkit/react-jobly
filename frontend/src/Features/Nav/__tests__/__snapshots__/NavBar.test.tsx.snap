// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-dark bg-secondary"
  >
    <div
      class="container-fluid"
    >
      <ul
        class="d-flex align-items-center nav"
      >
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="navbar-brand active"
            href="/"
          >
            React-Jobly
          </a>
        </li>
      </ul>
      <li
        class="nav-item"
        style="list-style: none;"
      >
        <a
          class=""
          href="/login"
        >
          Login
        </a>
        <a
          class=""
          href="/signup"
        >
          Sign Up
        </a>
      </li>
    </div>
  </nav>
</DocumentFragment>
`;
